### PR TITLE
deprecate 'frequencies' reconstructing_fractional_octave_bands()

### LINKED ISSUE #657: Inconsistency for fractional filter returns
[fractional_octave_bands](https://pyfar.readthedocs.io/en/stable/modules/pyfar.dsp.filter.html#pyfar.dsp.filter.fractional_octave_bands) and [reconstructing_fractional_octave_bands](https://pyfar.readthedocs.io/en/stable/modules/pyfar.dsp.filter.html#pyfar.dsp.filter.reconstructing_fractional_octave_bands) have different output parameter

seems like ``reconstructing_fractional_octave_bands``returns signal, frequencies or filter, frequencies
and ``fractional_octave_bands`` returns signal or filter

also the inputs are not perfect. the ``sampling_rate`` is not at the same position and ``num_fractions`` has sometimes defaults and another time not.
```
pyfar.dsp.filter.reconstructing_fractional_octave_bands(signal, num_fractions=1, frequency_range=(63, 16000), overlap=1, slope=0, n_samples=4096, sampling_rate=None)

pyfar.dsp.filter.fractional_octave_bands(signal, num_fractions, sampling_rate=None, frequency_range=(20.0, 20000.0), order=14)
```

### PR DESCRIPTION
**Pull request by hoyer-a**
deprecate `frequencies` return in `reconstructing_fractional_octave_bands()`

Closes #657

### CODE REVIEW

**Review by sikersten — APPROVED** [review]
Thanks for taking care!
Is there something we can do about the issue with the input parameters mentioned in #657?

**Comment by hoyer-a** [comment]
Ah, yes. All the other filters have `sampling_rate` as the last parameter. I can move the parameter in `fractional_octave_bands` to the end to match the rest of `dsp.filter`. Concerning the default for num_fractions, i think setting `num_fractions=1` in `fractional_octave_bands` would be ok too.

Any opinions and can i do it here or is it a beer?

**Review by f-brinkmann — DISMISSED** [review]

File: pyfar/dsp/filter/fractional_octaves.py [hunk 1]
```diff
@@ -396,7 +397,8 @@ def reconstructing_fractional_octave_bands(
     filter : FilterFIR
         FIR Filter object. Only returned if ``signal = None``.
     frequencies : np.ndarray
-        Center frequencies of the filters.
+        Center frequencies of the filters. ``'frequencies'`` return will be
+        deprecated in pyfar 0.9.0.
```
Comment (ahms5) [inline pyfar/dsp/filter/fractional_octaves.py:401]: maybe we can add what to use instead.

File: pyfar/dsp/filter/fractional_octaves.py [hunk 2]
```diff
@@ -518,6 +520,9 @@ def reconstructing_fractional_octave_bands(
         f"(num_fractions={num_fractions}, frequency_range={frequency_range}, "
         f"overlap={overlap}, slope={slope})")
 
+    warnings.warn(("Return parameter 'frequencies' will be deprecated in pyfar"
+                   " 0.9.0"), PyfarDeprecationWarning, stacklevel=2)
```
Comment (ahms5) [inline pyfar/dsp/filter/fractional_octaves.py:524]: see above

**Review by ahms5 — CHANGES_REQUESTED** [review]
thank you for taking care! just a minor thing.

File: pyfar/dsp/filter/fractional_octaves.py (hunk 1 above)
Comment (ahms5) [reply pyfar/dsp/filter/fractional_octaves.py:401]: yes

**Review by mberz — CHANGES_REQUESTED** [review]
I would also adapt the example to use the respective function for getting the center frequencies.

File: pyfar/dsp/filter/fractional_octaves.py [hunk 3]
```diff
@@ -414,7 +417,10 @@ def reconstructing_fractional_octave_bands(
         >>> import matplotlib.pyplot as plt
         >>> # generate data
         >>> x = pf.signals.impulse(2**12)
-        >>> y, f = pf.dsp.filter.reconstructing_fractional_octave_bands(x)
+        >>> y, _ = pf.dsp.filter.reconstructing_fractional_octave_bands(x)
```
Comment (mberz) [inline pyfar/dsp/filter/fractional_octaves.py:420]:
```suggestion
        >>> y = pf.dsp.filter.reconstructing_fractional_octave_bands(x)[0]
```

File: pyfar/dsp/filter/fractional_octaves.py [hunk 4]
```diff
@@ -414,7 +417,10 @@ def reconstructing_fractional_octave_bands(
...
+        >>> # get center frequencies
+        >>> _, f = pf.dsp.filter.fractional_octave_frequencies(
+        ...    frequency_range=(60, 16000))
```
Comment (mberz) [inline pyfar/dsp/filter/fractional_octaves.py:423]:
```suggestion
        >>> f = pf.dsp.filter.fractional_octave_frequencies(
        ...    frequency_range=(60, 16000))[1]
```

**Review by mberz — COMMENTED** [review]
Just to make it a bit cleaner. Otherwise looking good.

**Review by mberz — APPROVED** [review]
All good, thanks!

**Review by f-brinkmann — APPROVED** [review]

**Review by ahms5 — APPROVED** [review]

**Review by sikersten — APPROVED** [review]

**Pull request merged**